unicorn/src/zookeeper.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+* 2015+ Copyright (c) Anton Matveenko <antmat@yandex-team.ru>
+* All rights reserved.
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+
+#include "cocaine/zookeeper.hpp"
+#include "cocaine/traits/unicorn.hpp"
+
+#include <cocaine/dynamic.hpp>
+#include <cocaine/errors.hpp>
+
+namespace cocaine {
+namespace zookeeper {
+
+path_t
+path_parent(const path_t& path, unsigned int depth) {
+    if(depth == 0) {
+        return path;
+    }
+    size_t last_char = path.size();
```

Comment:
this one is unchanged(only moved ) in fact and I don't see any problem in case path.empty() == true
This can be simplified though.